File: j.cpp
#include<bits/stdc++.h>

using namespace std;

#define ll long long
#define PB push_back
#define PII par<int, int>
#define MP make_pair
#define all(x) x.begin(), x.end()
#define MSET() memset(x, y, sizeof(x))
#define F first
#define S second
#define maxn

int n, m;
int arr[20];
string num[20];
string ans;
int minn;

void dfs(int now, string val, int cnt)
{

    bitset<505> curr(val);
    bitset<505> numm(num[now]);

    //cout << now << ' ' << cnt << endl;
    //cout << curr << endl;
    //cout << numm << endl;


    if(val == ans)
        minn = min(minn, cnt);

    if(now == m)
        return;
    dfs(now+1, val, cnt);
    dfs(now+1, (curr|numm).to_string(), cnt+1);
}

int main(){

    int cases;
    cin >> cases;
    while(cases--){

        minn = 1e9;
        scanf("%d %d", &n, &m);

        ans = "";

        for(int i = 504; i >= 0; i--)
        {
            ans += "0";
        }
        for(int i = 504; i >= 504-n+1; i--)
        {
            ans[i] = '1';
        }



        //cout << ans << endl;

        string hold;


        for(int i = 0; i < m; i++)
        {

            cin >> num[i];
        }



        dfs(0, "0", 0);

        if(minn == 1e9)
            printf("-1\n");
        else
            printf("%d\n", minn);
    }

    return 0;
}

File: j.py
cases = int(input())

while cases > 0:

    arr = [[0 for i in range(10)],[0 for j in range(10)]]
    cnt = [[0 for i in range(10)],[0 for j in range(10)]]
    num = [0 for i in range(510)]

    n, m = input().split()
    n = int(n)
    m = int(m)

    for i in range(m):
        tmp = input()
        num[i+1] = int(tmp, 2)

    print(arr[m-1][m-1])
    print(arr.size)
    for i in range(m):
        arr[i][i] = num[i]
        cnt[i][i] = 1

    flag = False
    for i in range(1, m+1):
        for j in range(i+1, m+1):

            if j+1 > m+1:
                break


            arr[j][j+1] = arr[j+1][j] & arr[j][j-1]
            cnt[j][j+1] = cnt[j+1][j] + cnt[j][j-1] -1
            if arr[j][j+1] == 2 ** (n-1):
                flag = True
                break

        if flag:
            break


    print(cnt)


    cases-=1
